### ISSUE TITLE
Is it possible to make the vstest@2 task use the new version (or the old version)?

### ISSUE THREAD

**Issue by Evangelink** [microsoft/testfx#1553]
I have this same issue in my Azure Pipelines, using vstest@2, no direct mstest reference. It's coming from MSTestAdapter: 

`MSTestAdapter failed to discover tests in class`

I'm able to make the error stop by using the runsettings disablement of AppDomain, but the tests need the AppDomain, so we need another option.

Is it possible to make the vstest@2 task use the new version (or the old version)?

This is the task I'm referring to:

https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/vstest-v2?view=azure-pipelines

_Originally posted by @GaTechThomas in https://github.com/microsoft/testfx/issues/1493#issuecomment-1381310699_

**Comment by Evangelink** [comment]
Hi @GaTechThomas! Thanks for the bug report. Sadly this repository is not linked in anyway to vstest task. @dogirala Can you please let me know where users can report AzDo task issues?

**Comment by dogirala** [comment]
Hi @Evangelink, if the bug is realted to AzDo tasks, then I believe the right team would be Pipelines team, and <martinmrazik@microsoft.com> could help you I believe. If that didn't help, one other way I can suggest is to drop an email to create a customer query with Devops 'devops_tools@microsoft.com' who could help you with the issue reporting or maybe even triage it to the right team.

**Comment by cremor** [comment]
The repository for Azure DevOps pipeline tasks is here: https://github.com/microsoft/azure-pipelines-tasks
You could try filing an issue there.

**Comment by Evangelink** [comment]
@GaTechThomas I have just reviewed a little more your issue and I think I am a bit confused. VSTest task is not bringing any dependency to MSTest so I think it's a configuration of your solution/projects.

**Comment by GaTechThomas** [comment]
Thank you @Evangelink. The project that I've been seeing this on is massive and has a very large number of resources that it acquires from nuget, from binaries in the repo, and from tools installed on the pipeline agent machine. It's pretty difficult to sort out what bits are coming from where. As a result, I've shifted from working on this to working on a focused alternative that uses .NET 6 and proper package management.

In short, I'm on a path that likely will abandon the need associated with this ticket. I'll know more in the coming days. It's probably safe to go ahead and close this ticket.

**Comment by Evangelink** [comment]
Ok closing this ticket. If you decide to continue, I'd recommend you to create a ticket on https://developercommunity.visualstudio.com/AzureDevOps as it seems (with elements at hand) there is nothing to do in MSTest.
